Kernel/KBO.cpp

```diff
@@ -983,6 +983,7 @@ bool KboSpecialWeights<FuncSigTraits>::tryGetWeight(unsigned functor, unsigned&
 }
 
 template KboWeightMap<FuncSigTraits> KboWeightMap<FuncSigTraits>::dflt();
+template KboWeight KboWeightMap<FuncSigTraits>::symbolWeight(Term*) const;
```

Comment:
What do these actually mean? Are they necessary?

Reply:
@joe-hauns Martin has found that if comment out these template specialisations, everything still compiles fine. Do we really need them?

Reply:
So it is there to instantiate the `KboWeightMap<...>::symbolWeight` for `FuncSigTraits`. 
This means that the compiler will monomorphize and precompile it for `FuncSigTraits`, so you can have the implementation in a cpp (not a hpp) file and still use the template instance for `FuncSigTraits` in a different file. 
Probably this function is not used outside from the cpp file (anymore). This means that the explicit instantiation is not used anymore. 
Sooo if it compiles without this, just get rid of it yes :)